lib/diskoverflow/slice.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (C) 2018 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package diskoverflow
+
+import (
+	"fmt"
+)
+
+type Slice interface {
+	Append(v Value)
+	NewIterator() Iterator
+	NewReverseIterator() Iterator
+	Bytes() int
+	Items() int
+	SetOverflowBytes(bytes int)
+	Close()
```

Comment:
Iterators are nice and stuff, but I guess we could get away with indexing?
Then this whole  "iterating" nonsense goes away I guess?

Reply:
How do you do that in the kv-storage on disk?